src/main/groovy/life/qbic/registration/handler/RegistrationException.groovy

```diff
@@ -0,0 +1,37 @@
+package life.qbic.registration.handler
+
+/**
+ * <p>Dataset registration exception.</p>
+ * <br>
+ * <p>This exception indicates an issue during registration of datasets in openBIS.</p>
+ *
+ * @since 1.0.0
+ */
+class RegistrationException extends RuntimeException {
+
+    /**
+     * Default constructor.
+     */
+    RegistrationException() {
+        super()
+    }
```

Comment:
NIT: We could think about this being a chainable exception similiar to how the measurements are validated in the datamanager, which would avoid having to reregister the dataset multiple times for each occuring error 🤔